app/src/main/java/com/android/streetworkapp/ui/authentication/SignIn.kt

```diff
@@ -0,0 +1,131 @@
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.android.sample.R
+import com.android.streetworkapp.ui.navigation.NavigationActions
+import com.android.streetworkapp.ui.navigation.Screen
+import com.android.streetworkapp.utils.GoogleAuthService
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.ktx.Firebase
+
+@Composable
+fun SignInScreen(navigationActions: NavigationActions) {
+
+  // This part of the code handles google sign-in :
+  var user by remember { mutableStateOf(Firebase.auth.currentUser) }
+  val token = stringResource(R.string.default_web_client_id)
+  val context = LocalContext.current
+
+  // Create an instance of GoogleAuthService (helper class for authentication) :
+  val authService = remember { GoogleAuthService(token, Firebase.auth) }
+
+  // Instantiate the launcher for the sign-in process :
+
+  val launcher =
+      authService.rememberFirebaseAuthLauncher(
+          onAuthComplete = { result ->
+            user = result.user
+            Log.d("SignInScreen", "Sign-in successful user : $user")
+            Toast.makeText(context, "Login successful!", Toast.LENGTH_LONG).show()
+            navigationActions.navigateTo(Screen.MAP)
+          },
+          onAuthError = { user = null })
+
+  Box(modifier = Modifier.fillMaxSize().testTag("loginScreenBoxContainer")) {
+
+    // Centralized content
+    Column(
+        modifier = Modifier.fillMaxSize().testTag("loginScreenColumnContainer"),
+        verticalArrangement = Arrangement.Center,
+        horizontalAlignment = Alignment.CenterHorizontally) {
+          // Welcome text in the center
+          Text(
+              text = "Welcome to the Street Work'App",
+              style = TextStyle(fontSize = 24.sp),
+              modifier = Modifier.testTag("loginTitle"))
+
+          Spacer(modifier = Modifier.height(64.dp).testTag("loginScreenSpacer"))
+
+          // Google Sign-in button
+          Button(
+              onClick = {
+                Log.d("SignInScreen", "Start sign-in")
+                authService.launchSignIn(context, launcher)
+              },
+              modifier =
+                  Modifier.width(250.dp)
+                      .height(40.dp)
+                      .border(
+                          width = 1.dp,
+                          color = Color(0xFFDADCE0),
```

Comment:
Suggestion: don't use a hardcoded color and define it in a color palette somewhere with a proper name

Reply:
This is a great remark!

Reply:
This was corrected in recent fix of using the Color.kt file of the project